src/transCore.tsx

```diff
@@ -80,16 +81,35 @@ function getDicValue(
 ): string | undefined | object {
   const value: string | object = key
     .split('.')
-    .reduce((val: I18nDictionary | string, key: string) => {
-      if (typeof val === 'string') {
-        return {}
-      }
-
-      const res = val[key as keyof typeof val]
-
-      // pass all truthy values or (empty) strings
-      return res || (typeof res === 'string' ? res : {})
-    }, dic)
+    .reduce(
+      (
+        val: I18nDictionary | string | Container<I18nDictionary | string>,
+        key: string
+      ) => {
+        if (typeof val === 'string') {
+          return {}
+        }
+
+        /**
+         * Check if value is instance of container
+         */
+        const isContainer = <T,>(
+          val: string | I18nDictionary | Container<T>
+        ): val is Container<T> => {
+          return Array.isArray(val)
+        }
+
+        if (isContainer<I18nDictionary | string>(val)) {
+          return {}
```

Comment:
What is the purpose of this? It would be great to add a test.

Reply:
@lorenzosignoretti ping on this. It would be nice to merge this PR for the next release. What I don't understand is why it can't be solved just with TypeScript types and requires touching the logic. Would you clarify it for me? Thanks!